R/lockbox.R

```diff
@@ -122,3 +144,11 @@ disallow_special_packages <- function(lock) {
       ".", call. = FALSE)
   }
 }
+
+disallow_duplicate_packages <- function(lock) {
+  locked_names <- vapply(lock, function(p) p$name, character(1))
+  if (any(duplicated(locked_names))) {
+    stop(paste0("The following packages are duplicated in your lockfile: "
+      , paste(locked_names[duplicated(locked_names)], collapse = ", ")))
```

Comment:
`unique()` around this so you don't print the names multiple times in case of 3+ duplications